google-cloud-pub-sub/src/main/scala/akka/stream/alpakka/googlecloud/pubsub/GrpcApi.scala

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright (C) 2016-2018 Lightbend Inc. <http://www.lightbend.com>
+ */
+
+package akka.stream.alpakka.googlecloud.pubsub
+
+import akka.stream.Materializer
+
+/** Pub/Sub **/
+import com.google.auth.oauth2.GoogleCredentials
+import com.google.pubsub.v1.{PullRequest, SubscriberGrpc, PublisherGrpc, AcknowledgeRequest => AR}
+import com.google.pubsub.v1
+
+/** gRPC **/
+import akka.stream.alpakka.googlecloud.pubsub.GuavaConversions._
+import io.grpc.auth.MoreCallCredentials
+import io.grpc.{ManagedChannel, ManagedChannelBuilder, _}
+
+import scala.collection.JavaConverters._
+import scala.concurrent.Future
+
+@akka.annotation.InternalApi
+private class GrpcApi(project: String, subscription: String, config: PubSubConfig)(
+    implicit materializer: Materializer
+) {
+  import materializer.executionContext
+
+  type AckId = String
+
+  private val subscriptionFqrn = s"projects/$project/subscriptions/$subscription"
+
+  private val grpcChannel: ManagedChannel = {
+    val builder = ManagedChannelBuilder.forAddress(config.host, config.port)
+    if (config.usePlaintext) builder.usePlaintext(true)
+    builder.build()
+  }
+
+  private val callCredentials: CallCredentials =
```

Comment:
It would be extremaly usefull to allow providing Credentials from the outside, eg. `NoCredentialsProvider.create` for the Emulator.

Reply:
You are not using builders which accept `CredentialsProvider` but `CallCredentials` directly, is there any reason for that? 
When using API that way to add support for NoCredentialsProvider you would have to skip `withCallCredentials` call when provided credentials are `null`/`None`.